### PR TITLE
#256 Adding Github Actions NPM Publishing

### PR DESCRIPTION
**Pull request by erikmjohnson**
Added npm publishing in github action.

Needs token and pathway

### CODE REVIEW

File: .github/workflows/npmpublish.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      - name: Install
+        run: npm ci
+      - name: Publish
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
+        run: npm publish
+
+  publish-gpr:
```
Comment (anthonyjgrove) [inline .github/workflows/npmpublish.yml]: Can remove `publish-gpr` section

File: .github/workflows/npmpublish.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+name: Node.js Package
+
+on:
+  release:
+    types: [created]
```
Comment (anthonyjgrove) [inline .github/workflows/npmpublish.yml]: want to scope this to only run on tags starting with `v`

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: Publish
```
Comment (anthonyjgrove) [inline .github/workflows/publish.yml]: Rename this to Publish Storybook, same with the filename

File: .github/workflows/npmpublish.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 12
+          registry-url: https://registry.npmjs.org/
+      - name: Install
+        run: npm ci
```
Comment (anthonyjgrove) [inline .github/workflows/npmpublish.yml]: need bundle step as well

File: package.json [hunk 5]
```diff
@@ -44,7 +44,7 @@
   },
   "homepage": "https://github.com/anthonyjgrove/react-google-login",
   "dependencies": {
-    "@types/react": "*",
+    "@types/react": "^16.9.11",
```
Comment (anthonyjgrove) [inline package.json]: why did you change the `*` ?

File: package.json [hunk 6]
```diff
@@ -61,13 +61,15 @@
...
     "@storybook/storybook-deployer": "^2.8.1",
-    "autoprefixer": "^9.5.0",
+    "autoprefixer": "^9.7.1",
     "babel-core": "7.0.0-bridge.0",
     "babel-jest": "^24.7.1",
     "babel-loader": "^8.0.5",
     "babel-plugin-transform-react-constant-elements": "^6.23.0",
     "babel-plugin-transform-react-inline-elements": "^6.22.0",
     "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
+    "child_process": "^1.0.2",
```
Comment (anthonyjgrove) [inline package.json]: why did these get added? `child_process` and `dotenv`?

File: webpack/webpack.config.dev.js [hunk 7]
```diff
@@ -44,7 +44,9 @@ module.exports = {
   plugins: [
     new webpack.HotModuleReplacementPlugin(),
     new webpack.DefinePlugin({
-      'process.env.NODE_ENV': JSON.stringify('development')
+      'process.env': {
```
Comment (anthonyjgrove) [inline webpack/webpack.config.dev.js]: why did you change this?

File: webpack/webpack.config.prod.js [hunk 8]
```diff
@@ -50,7 +50,9 @@ module.exports = {
   },
   plugins: [
     new webpack.DefinePlugin({
-      'process.env.NODE_ENV': JSON.stringify('production')
+      'process.env': {
```
Comment (anthonyjgrove) [inline webpack/webpack.config.prod.js]: Same here, why did this get changed?

File: .github/workflows/npmpublish.yml (hunk 2 above)
Comment (erikmjohnson) [reply .github/workflows/npmpublish.yml]: Added to file

File: .github/workflows/npmpublish.yml (hunk 4 above)
Comment (erikmjohnson) [reply .github/workflows/npmpublish.yml]: added

File: .github/workflows/npmpublish.yml (hunk 1 above)
Comment (erikmjohnson) [reply .github/workflows/npmpublish.yml]: removed

File: .github/workflows/publish.yml (hunk 3 above)
Comment (erikmjohnson) [reply .github/workflows/publish.yml]: Name changed

File: package.json (hunk 5 above)
Comment (erikmjohnson) [reply package.json]: Not sure how that happened, I reverted it back

File: package.json (hunk 6 above)
Comment (erikmjohnson) [reply package.json]: Sorry, the rest of the notes below are related to me trying to figure out how to publish storybook through github actions. Everything is reverted.

File: webpack/webpack.config.dev.js (hunk 7 above)
Comment (erikmjohnson) [reply webpack/webpack.config.dev.js]: See notes on child_process and env

File: webpack/webpack.config.prod.js (hunk 8 above)
Comment (erikmjohnson) [reply webpack/webpack.config.prod.js]: See notes on child_process and env
